torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -1252,16 +1253,24 @@ xla::XlaOp BuildCustomSharding(const xla::XlaOp& input) {
 xla::XlaOp BuildTpuCustomCall(const std::vector<xla::XlaOp>& inputs,
                               const xla::Shape& output_shape,
                               const std::string& payload) {
+  // We need to enforce the default C-order (major-to-minor) layouts for inputs
+  // to Mosaic and outputs from Mosaic.
   std::vector<xla::Shape> input_shapes;
   input_shapes.reserve(inputs.size());
   for (const auto& input : inputs) {
-    input_shapes.push_back(ShapeHelper::ShapeOfXlaOp(input));
+    auto shape = ShapeHelper::ShapeOfXlaOp(input);
+    input_shapes.push_back(MakeTorchTensorLayout(
+        shape.dimensions(), shape.dynamic_dimensions(), shape.element_type()));
   }
+  auto output_shape_impl = MakeTorchTensorLayout(
```

Comment:
lol can you spell out the `xla::Shape` directly here? I need to do a search to figure out the return type of `MakeTorchTensorLayout `. Same for above.

I tried to follow Google's C++ recommendation which states
```
Type Deduction (including auto)
Use type deduction only if it makes the code clearer to readers who aren't familiar with the project, or if it makes the code safer. Do not use it merely to avoid the inconvenience of writing an explicit type.
```